### PR TITLE
Add support for custom palette in editor (#315)

### PR DESCRIPTION
**Pull request by merwok**
Ref #315
Alternative to #301

### CODE REVIEW

File: pyxel/editor/__init__.py [hunk 1]
```diff
@@ -6,19 +6,29 @@
...
 def run():
-    arg = sys.argv[1] if len(sys.argv) >= 2 else ""
+    palette = None
+    filename = "my_resource"
 
-    if arg.startswith("-"):
-        if arg == "-v" or arg == "--version":
-            print("Pyxel Editor {}".format(pyxel.VERSION))
+    for arg in sys.argv[1:]:
+        if arg.startswith("-"):
```
Comment (merwok) [inline pyxel/editor/__init__.py:13]: At this point we could use `argparse`.

File: README.md [hunk 2]
```diff
@@ -255,6 +255,15 @@ After starting Pyxel Editor, the file can be switched by dragging and dropping a
 
 The created resource file can be loaded with the `load` function.
 
+If your game uses a custom palette, you can use it in the editor with this option:
+
+```sh
+pyxeleditor --palette=path/to/palette [pyxel_resource_file]
+```
+
+The palette file must be a text file containing 16 color codes (like `1D2B53`) on 16 lines.
```
Comment (merwok) [inline README.md:264]: This is the `hex` format on Lospec.  Could also allow a leading `#` easily.

File: pyxel/editor/__init__.py [hunk 3]
```diff
@@ -6,19 +6,29 @@
...
+                print("Usage: pyxeleditor [option] [pyxel_resource_file]")
+                print("Options:")
+                print(" -h, --help     This help text")
+                print(" -v, --version  Show version number and quit")
+                print(" --palette file Load custom palette from file")
+                return
+            elif arg.startswith("--palette="):
+                palette_file = arg[len("--palette="):]
+                with open(palette_file) as file:
+                    palette = [int(line.lstrip("#"), 16) for line in file.read().splitlines()]
```
Comment (merwok) [inline pyxel/editor/__init__.py:27]: I could improve error reporting in case of bad file format.

**Comment by merwok** [comment]
I have a branch that merges develop, but it’s untested (don’t want to install rustup now).

**Comment by merwok** [comment]
I have merged develop but not tested it (don’t want to install rustup now).

I see now that I have to merge again, probably because of the force pushes on develop 🙁

**Comment by kitao** [comment]
All code has been reimplemented

**Pull request closed without merging**